src/tests/function/left_right.rs

```diff
@@ -0,0 +1,179 @@
+use crate::*;
+
+test_case!(left_right, async move {
+    use Value::Str;
+    let test_cases = vec![
+        ("CREATE TABLE Item (name TEXT)", Ok(Payload::Create)),
+        (
+            r#"INSERT INTO Item VALUES ("Blop mc blee"), ("B"), ("Steven the &long named$ folken!")"#,
+            Ok(Payload::Insert(3)),
+        ),
+        (
+            "CREATE TABLE SingleItem (id INTEGER PRIMARY KEY)",
+            Ok(Payload::Create),
+        ),
+        (
+            r#"INSERT INTO SingleItem VALUES (0)"#,
+            Ok(Payload::Insert(1)),
+        ),
+        (
+            "CREATE TABLE NullName (name TEXT NULL)",
+            Ok(Payload::Create),
+        ),
+        (
+            r#"INSERT INTO NullName VALUES (NULL)"#,
+            Ok(Payload::Insert(1)),
+        ),
+        (
+            "CREATE TABLE NullNumber (number INTEGER NULL)",
+            Ok(Payload::Create),
+        ),
+        (
+            r#"INSERT INTO NullNumber VALUES (NULL)"#,
+            Ok(Payload::Insert(1)),
+        ),
+        (
+            "CREATE TABLE NullableName (name TEXT NULL)",
+            Ok(Payload::Create),
+        ),
+        (
+            r#"INSERT INTO NullableName VALUES ('name')"#,
+            Ok(Payload::Insert(1)),
+        ),
+        (
+            r#"SELECT LEFT(name, 3) AS test FROM Item"#,
+            Ok(select!(
+                "test"
+                Str;
+                "Blo".to_owned();
+                "B".to_owned();
+                "Ste".to_owned()
+            )),
+        ),
+        (
+            r#"SELECT RIGHT(name, 10) AS test FROM Item"#,
+            Ok(select!(
+                "test"
+                Str;
+                "op mc blee".to_owned();
+                "B".to_owned();
+                "d$ folken!".to_owned()
+            )),
+        ),
+        // TODO Concatenation
+        /*(
+            r#"SELECT LEFT((name + 'bobbert'), 10) AS test FROM Item"#,
+            Ok(select!(
+                "test"
+                Str;
+                "Blop mc blee".to_owned();
+                "Bbobbert".to_owned();
+                "Steven the".to_owned()
+            )),
+        ),*/
+        (
+            r#"SELECT LEFT('blue', 10) AS test FROM SingleItem"#,
+            Ok(select!(
+                "test"
+                Str;
+                "blue".to_owned()
+            )),
+        ),
+        (
+            r#"SELECT LEFT("blunder", 3) AS test FROM SingleItem"#,
+            Ok(select!(
+                "test"
+                Str;
+                "blu".to_owned()
+            )),
+        ),
+        (
+            r#"SELECT LEFT(name, 3) AS test FROM NullName"#,
+            Ok(select!(
+                "test"
+                Str;
+                "".to_owned()
+            )),
```

Comment:
Here, the type of `name` is `OptStr`, then I think it's better to keep `OptStr` and keep `None`. How do you think of this?
This `LEFT` converted `NULL` to empty string.

Reply:
Ah good point, thanks, was just copy pasting :-)

Reply:
Oh oops now I see what you mean, yeah. Thanks.